test/chi/custom-elements/buttons.pug

```diff
@@ -0,0 +1,197 @@
+                      disabled= state === 'disabled'
+                      extra-class= `-${state}`
+                    )= 'Button'
+                    chi-button.-m--1(
+                      type='icon'
+                      color=style
+                      size= size !== 'default' ? size : ''
+                      outline= type === 'outline'
+                      flat= type === 'flat'
+                      fluid= type === 'fluid'
+                      disabled= state === 'disabled'
+                      extra-class= `-${state}`
+                    )
+                      .a-btn__content
+                        chi-icon(icon='atom')
+                    chi-button.-m--1(
+                      color=style
+                      size= size !== 'default' ? size : ''
+                      outline= type === 'outline'
+                      flat= type === 'flat'
+                      fluid= type === 'fluid'
+                      disabled= state === 'disabled'
+                      extra-class= `-${state}`
+                    )
+                      .a-btn__content
+                        chi-icon(icon='atom')
+                        span Button
+                    chi-button.-m--1.-ml--2(
+                      color=style
+                      size= size !== 'default' ? size : ''
+                      outline= type === 'outline'
+                      flat= type === 'flat'
+                      fluid= type === 'fluid'
+                      pill
+                      disabled= state === 'disabled'
+                      extra-class= `-${state}`
+                    )
+                      .a-btn__content
+                        span Button
+                    chi-button.-m--1(
+                      color=style
+                      size= size !== 'default' ? size : ''
+                      type='icon'
+                      outline= type === 'outline'
+                      flat= type === 'flat'
+                      fluid= type === 'fluid'
+                      pill
+                      disabled= state === 'disabled'
+                      extra-class= `-${state}`
+                    )
+                      .a-btn__content
+                        chi-icon(icon='atom')
+                    chi-button.-m--1(
+                      color=style
+                      size= size !== 'default' ? size : ''
+                      pill
+                      outline= type === 'outline'
+                      flat= type === 'flat'
+                      fluid= type === 'fluid'
+                      disabled= state === 'disabled'
+                      extra-class= `-${state}`
+                    )
+                      .a-btn__content
+                        chi-icon(icon='atom')
+                        span Button
+              else
+                td
+                  .-mr--2
+                    chi-button.-m--1.-ml--2(
+                      data-cy="button"
```

Comment:
@srinath-koneru This selector is very generic and repetitive, so it won't be useful in the test files. button-primary-pill-xl would be more appropriate (change as necessary).